app/components/Policies/Manage.jsx

```diff
@@ -93,86 +102,79 @@ export default class Manage extends React.Component {
+                <JsonEditor
+                    rootName={this.state.focusPolicy}
+                    value={this.state.currentPolicy}
+                    mode={'code'}
+                    schema={jsonschema}
+                    onChange={policyChanged}
+                />
             </Dialog>
         );
     }
 
     renderNewPolicyDialog() {
         const MISSING_POLICY_ERROR = "Policy cannot be empty.";
-        const DUPLICATE_POLICY_ERROR = `Policy ${this.state.newPolicy.name} already exists.`;
+        const DUPLICATE_POLICY_ERROR = `Policy ${this.state.focusPolicy} already exists.`;
 
         let validateAndSubmit = () => {
-            if (this.state.newPolicy.name === '') {
+            if (this.state.focusPolicy === '') {
                 this.setState({
                     newPolicyErrorMessage: MISSING_POLICY_ERROR
                 });
                 return;
             }
 
-            if (_.filter(this.state.policies, x => x.name === this.state.newPolicy.name).length > 0) {
+            if (_.filter(this.state.policies, x => x.name === this.state.focusPolicy).length > 0) {
                 this.setState({
                     newPolicyErrorMessage: DUPLICATE_POLICY_ERROR
                 });
                 return;
             }
-
-            axios.put(`/policy?vaultaddr=${encodeURI(window.localStorage.getItem("vaultUrl"))}&policy=${encodeURI(this.state.newPolicy.name)}&token=${encodeURI(window.localStorage.getItem("vaultAccessToken"))}`, { "Policy": this.state.currentPolicy })
-                .then((resp) => {
-                    let policies = this.state.policies;
-                    policies.push({ name: this.state.newPolicy.name });
-                    this.setState({
-                        policies: policies,
-                        errorMessage: ''
-                    });
-                })
-                .catch((err) => {
-                    console.error(err.stack);
-                    this.setState({
-                        errorMessage: err.response.data
-                    });
-                })
-
-            this.setState({ openNewPolicyModal: false });
+            this.updatePolicy(this.state.focusPolicy, true);
         }
 
         const actions = [
             <FlatButton label="Cancel" primary={true} onTouchTap={() => this.setState({ openNewPolicyModal: false, newPolicyErrorMessage: '' })} />,
-            <FlatButton label="Submit" primary={true} onTouchTap={validateAndSubmit} />
+            <FlatButton label="Submit" disabled={this.state.disableSubmit} primary={true} onTouchTap={validateAndSubmit} />
         ];
 
-        let setNewPolicy = (e, v) => {
-            let currentPolicy = this.state.newPolicy;
-            if (e.target.name === "newName") {
-                currentPolicy.name = v;
-            } else if (e.target.name === "newRules") {
-                currentPolicy.rules = v;
+        let setNewPolicy = (v, syntaxCheckOk, schemaCheckOk) => {
```

Comment:
This looks like it should be its own function, since you are reusing the same logic on [L108](https://github.com/djenriquez/vault-ui/pull/38/files#diff-5468412a869961d3e4754e751324bd06R108)